Visualisation of the videos

Ihashi opened the pull request:
Buttons in order to change the video

Ihashi committed:
Visualisation of the videos
Buttons in order to change the video

Ihashi committed:
Add the new videos

Ihashi committed:
Loop the videos

Ihashi committed:
Correction of the bug (loop video)
Few minor changes

jdauphant commented on MiniProject 1/VideoViewController.swift:
```diff
@@ -0,0 +1,115 @@
+        moviePlayer!.actionAtItemEnd = .None
+        
+        playerLayer!.frame = CGRect(x: 0, y: 0, width: screenSize.height, height: screenSize.width)
+        //playerLayer!.backgroundColor = UIColor.blackColor().CGColor
```
Do not commit commented code.

jdauphant commented on MiniProject 1/VideoViewController.swift:
```diff
@@ -0,0 +1,115 @@
+        playerItem = AVPlayerItem(URL: url)
+        moviePlayer = AVPlayer(playerItem: playerItem)
+        playerLayer = AVPlayerLayer(player: moviePlayer)
+        moviePlayer!.actionAtItemEnd = .None
```
Do not force optionnal, use "let if" or myObject? instead for a crash proof code.

jdauphant commented on MiniProject 1/VideoViewController.swift:
```diff
@@ -0,0 +1,115 @@
+    
+    func displayButton()
+    {
+        let leftButton = UIButton.buttonWithType(UIButtonType.System) as! UIButton
```
Why the button haven't been added in the storyboard ?

jdauphant replied:
You can use "as?" instead of "as!" for crash proof code

jdauphant commented on MiniProject 1/VideoViewController.swift:
```diff
@@ -0,0 +1,115 @@
+    
+    func leftButtonAction(sender:UIButton!)
+    {
+        moviePlayer!.pause()
```
Do not force optionnal, use "if let" or myObject?myMethods() instead for a crash proof code.

jdauphant commented on MiniProject 1/VideoViewController.swift:
```diff
@@ -0,0 +1,115 @@
+    
+    func rightButtonAction(sender:UIButton!)
+    {
+        moviePlayer!.pause()
```
Do not force optionnal, use "let if" or myObject?myMethods() instead for a crash proof code.

jdauphant commented on MiniProject 1/VideoViewController.swift:
```diff
@@ -0,0 +1,115 @@
+        playerLayer = AVPlayerLayer(player: moviePlayer)
+        moviePlayer!.actionAtItemEnd = .None
+        
+        playerLayer!.frame = CGRect(x: 0, y: 0, width: screenSize.height, height: screenSize.width)
```
Do not force optionnal, use "let if" or myObject? instead for a crash proof code.

jdauphant commented on MiniProject 1/VideoViewController.swift:
```diff
@@ -0,0 +1,115 @@
+        self.view.backgroundColor = .blackColor()
+        self.view.layer.addSublayer(playerLayer)
+        
+        moviePlayer!.play()
```
Do not force optionnal, use "let if" or myObject? instead for a crash proof code.

jdauphant commented on MiniProject 1/VideoViewController.swift:
```diff
@@ -0,0 +1,115 @@
+        NSNotificationCenter.defaultCenter().addObserver(self,
+            selector: "restartVideo",
+            name: AVPlayerItemDidPlayToEndTimeNotification,
+            object: moviePlayer!.currentItem)
```
You have to cleanup the notification when the object "moviePlayer!.currentItem" or self is destroyed.
Take a look at deinit().

jdauphant commented on MiniProject 1/VideoViewController.swift:
```diff
@@ -0,0 +1,115 @@
+        let preferredTimeScale : Int32 = 1
+        let seekTime : CMTime = CMTimeMake(seconds, preferredTimeScale)
+        
+        moviePlayer!.seekToTime(seekTime)
```
Same comment for optionnal

jdauphant commented on MiniProject 1/VideoViewController.swift:
```diff
@@ -0,0 +1,115 @@
+        let seekTime : CMTime = CMTimeMake(seconds, preferredTimeScale)
+        
+        moviePlayer!.seekToTime(seekTime)
+        moviePlayer!.play()
```
Same comment for optionnal

jdauphant commented on MiniProject 1/VideoViewController.swift:
```diff
@@ -0,0 +1,115 @@
+    var urls = ["http://fun.siz.io/stories/142893791787803c7fb48f4d/0.mp4",
+                "http://fun.siz.io/stories/142893791787803c7fb48f4d/1.mp4",
+                "http://fun.siz.io/stories/142893791787803c7fb48f4d/2.mp4",
+                "http://fun.siz.io/stories/142893791787803c7fb48f4d/3.mp4"]
```
I forgot to tell you but it's will be better if all videos are differents (and not from the same story).

jdauphant commented:
The view is not correct on iOS 8.
On iOS 7 : screenSize.height and  screenSize.width are not oriented
On iOS 8 : screenSize.height and  screenSize.width take care of the screen orientation

jdauphant commented on MiniProject 1/VideoViewController.swift:
```diff
@@ -0,0 +1,115 @@
+        self.view.addSubview(rightButton)
+    }
+    
+    func leftButtonAction(sender:UIButton!)
```
This method need the action name and don't need argument.

jdauphant commented on MiniProject 1/VideoViewController.swift:
```diff
@@ -0,0 +1,115 @@
+
+class VideoViewController: UIViewController
+{
+    var moviePlayer: AVPlayer?
```
This var should be private.

jdauphant commented on MiniProject 1/VideoViewController.swift:
```diff
@@ -0,0 +1,115 @@
+        rightButton.addTarget(self, action: "rightButtonAction:", forControlEvents: UIControlEvents.TouchUpInside)
+        
+        self.view.addSubview(leftButton)
+        self.view.addSubview(rightButton)
```
Use self only when needed.

jdauphant commented:
You can use this AVPlayerView :

``` swift
import UIKit
import AVFoundation

class AVPlayerView : UIView {

    override class func layerClass() -> AnyClass {
        return AVPlayerLayer.self
    }

    var videoPlayer: AVPlayer? {
        set {
            let layer: AVPlayerLayer = self.layer as! AVPlayerLayer
            layer.player = newValue
            layer.videoGravity = AVLayerVideoGravityResizeAspect
        }
        get {
            let layer: AVPlayerLayer = self.layer as! AVPlayerLayer
            return layer.player
        }
    }
}
```

Ihashi committed:
Correction after the review
Not forced optionnal
Cleanup the notification
Different videos
Private variables

Ihashi committed:
Use of an UIView called AVPlayerView
Suppression of the buttons and view

jdauphant commented on MiniProject 1/VideoViewController.swift:
```diff
@@ -0,0 +1,95 @@
+    
+    @IBAction func mainButtons(sender: UIButton)
+    {
+        let buttonName = sender.currentTitle!
```
You are forcing a optionnal

jdauphant commented on MiniProject 1/VideoViewController.swift:
```diff
@@ -0,0 +1,95 @@
+                "http://fun.siz.io/stories/1429021517853eac7ce1fc29/0.mp4",
+                "http://fun.siz.io/stories/1429018976114f45db3a5f88/0.mp4"]
+    
+    @IBAction func mainButtons(sender: UIButton)
```
A another solution could have been 2 differents methods "prev" and "next".

jdauphant replied:
No intervention needed

jdauphant commented on MiniProject 1/VideoViewController.swift:
```diff
@@ -0,0 +1,95 @@
+        videoPlayer = AVPlayer(playerItem: playerItem)
+        videoPlayer?.actionAtItemEnd = .None
+        
+        self.playerView.setPlayer(self.videoPlayer!)
```
No need to use self.
You force a optional.

jdauphant commented on MiniProject 1/VideoViewController.swift:
```diff
@@ -0,0 +1,95 @@
+        videoPlayer?.actionAtItemEnd = .None
+        
+        self.playerView.setPlayer(self.videoPlayer!)
+        self.playerView.setVideoFillMode(AVLayerVideoGravityResizeAspect)
```
No need to use self.

jdauphant commented on MiniProject 1/VideoViewController.swift:
```diff
@@ -0,0 +1,95 @@
+        NSNotificationCenter.defaultCenter().addObserver(self,
+            selector: "restartVideo",
+            name: AVPlayerItemDidPlayToEndTimeNotification,
+            object: videoPlayer?.currentItem)
```
You don't clean the notification when "videoPlayer?.currentItem" or "self" are destroy. (already on the last code review)
Follow the (*)

jdauphant commented on MiniProject 1/VideoViewController.swift:
```diff
@@ -0,0 +1,95 @@
+    {
+        let url = NSURL(string: urls[videoNumber])
+        playerItem = AVPlayerItem(URL: url)
+        videoPlayer = AVPlayer(playerItem: playerItem)
```
It's preferable to use "if let" for playerItem and videoPlayer.

jdauphant commented on MiniProject 1/Base.lproj/Main.storyboard:
```diff
@@ -1,25 +1,76 @@
+            <point key="canvasLocation" x="408" y="382"/>
         </scene>
     </scenes>
 </document>
```
PREV and NEXT button should be placed as child of the "View" and not as child of the "Player View"
Constraint of the Player View are not clean (remove magic number) and introduce black border.

jdauphant commented on MiniProject 1/VideoViewController.swift:
```diff
@@ -0,0 +1,95 @@
+    func playVideo()
+    {
+        let url = NSURL(string: urls[videoNumber])
+        playerItem = AVPlayerItem(URL: url)
```
(*)

Ihashi committed:
Correction after the second review
Remote the notification
Not forced optionnal
